pytorch_lightning/trainer/training_loop.py

```diff
@@ -834,13 +837,19 @@ def run_on_epoch_end_hook(self, epoch_output):
         self.trainer.call_hook("on_train_epoch_end", epoch_output)
 
     def increment_accumulated_grad_global_step(self):
-        num_accumulated_batches_reached = (self.trainer.batch_idx + 1) % self.trainer.accumulate_grad_batches == 0
-        num_training_batches_reached = (self.trainer.batch_idx + 1) == self.trainer.num_training_batches
+        num_accumulated_batches_reached = self.accumulated_batches_reached()
+        num_training_batches_reached = self.num_training_batches_reached()
 
         # progress global step according to grads progress
         if num_accumulated_batches_reached or num_training_batches_reached:
             self.trainer.global_step += 1
 
+    def accumulated_batches_reached(self):
```

Comment:
can we have it rather as protected as it is not meant to be used by the user...

Reply:
Will make a followup PR